Core/Src/main.c

```diff
@@ -87,6 +94,8 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
```

Comment:
One thing you are probably not aware of, you need to call `HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);` to get pwm generated from the timer.